include/elf_parser.hpp

```diff
@@ -0,0 +1,191 @@
+    std::println("entry: 0x{:X}", m_ehdr.e_entry);
+    std::println("phoff: 0x{:X}", m_ehdr.e_phoff);
+    std::println("shoff: 0x{:X}", m_ehdr.e_shoff);
+    std::println("flags: 0x{:X}", m_ehdr.e_flags);
+    std::println("ehsize: 0x{:X}", m_ehdr.e_ehsize);
+    std::println("phentsize: 0x{:X}", m_ehdr.e_phentsize);
+    std::println("shentsize: 0x{:X}", m_ehdr.e_shentsize);
+    std::println("shnum: 0x{:X}", m_ehdr.e_shnum);
+    std::println("phnum: 0x{:X}", m_ehdr.e_phnum);
+}
+
+void ElfParser::printPhdr()
+{
+    if (elf_getphdrnum(m_e, &m_n) != 0) {
+        errx(EX_SOFTWARE, "getphdrnum() failed: %s.", elf_errmsg(-1));
+    }
+
+    std::println("Program Header: ({})", static_cast<int>(m_n));
+    std::println("====================");
+
+    for (int i = 0; i < static_cast<int>(m_n); i++) {
+        if (gelf_getphdr(m_e, i, &m_phdr) != &m_phdr) {
+            errx(EX_SOFTWARE, "getphdr() failed: %s.", elf_errmsg(-1));
+        }
+
+        std::println("Program Header: {}", i);
+        std::println("type : 0x{:X}", m_phdr.p_type);
+        std::println("flags : 0x{:X}", m_phdr.p_flags);
+        std::println("offset : 0x{:X}", m_phdr.p_offset);
+        std::println("vaddr : 0x{:X}", m_phdr.p_vaddr);
+        std::println("paddr : 0x{:X}", m_phdr.p_paddr);
+        std::println("filez : 0x{:X}", m_phdr.p_filesz);
+        std::println("memz : 0x{:X}", m_phdr.p_memsz);
+        std::println("align : 0x{:X}", m_phdr.p_align);
+        std::println("====================");
+    }
+}
+
+void ElfParser::printShdr()
+{
+    if (elf_getshdrstrndx(m_e, &m_shstrndx) != 0) {
+        errx(EX_SOFTWARE, "getshdrstrndx() failed: %s.", elf_errmsg(-1));
+    }
+
+    m_scn = NULL;
+
+    std::println("Section Header: ({})", static_cast<int>(m_shstrndx));
+    std::println("====================");
+
+    while ((m_scn = elf_nextscn(m_e, m_scn)) != NULL) {
+        if (gelf_getshdr(m_scn, &m_shdr) != &m_shdr) {
+            errx(EX_SOFTWARE, "getshdr() failed: %s.", elf_errmsg(-1));
+        }
+
+        if ((m_name = elf_strptr(m_e, m_shstrndx, m_shdr.sh_name)) == NULL) {
+            errx(EX_SOFTWARE, "elf_strptr() failed: %s.", elf_errmsg(-1));
+        }
+
+        std::println(
+          "Section {}: {}", static_cast<uintmax_t>(elf_ndxscn(m_scn)), m_name);
+        std::println("   type : 0x{:X}", m_shdr.sh_type);
+        std::println("   flags : 0x{:X}", m_shdr.sh_flags);
+        std::println("   addr : 0x{:X}", m_shdr.sh_addr);
+        std::println("   offset : 0x{:X}", m_shdr.sh_offset);
+        std::println("   size : 0x{:X}", m_shdr.sh_size);
+        std::println("   link : 0x{:X}", m_shdr.sh_link);
+        std::println("   info : 0x{:X}", m_shdr.sh_info);
+        std::println("   addralign : 0x{:X}", m_shdr.sh_addralign);
+        std::println("   entsize : 0x{:X}", m_shdr.sh_entsize);
+        std::println("====================");
+    }
+}
+
+void ElfParser::getSection(std::string section)
```

Comment:
```suggestion
void ElfParser::getSection(std::string const& section)
```
This way we don't do a copy and your function doesn't modify the value so we should be const correct here as well.